### DISCUSSION TITLE
Changeto not triggering on visible locations

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14341] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Initially posted in https://github.com/FakeFishGames/Barotrauma/discussions/14315#discussion-6955505

Locations that are clearly visible on the map fail to apply Changeto, leading to improper level generation.

In the video example, using Dynamic Europa, any outpost (here a hospital) is set to immedately turn into abandoned if adjacent to a Lair. After jumping using cheats (I did this multiple times previously), the location does not turn until I jump *past it*, pushing back the fog of war

```xml
      <ChangeTo type="Abandoned" locationtypechangecooldown="2" messagetag="locationchange.outpost.changeto.abandoned" requirediscovered="false">
        <Requirement requiredlocations="Lair,LairMudraptor,LairThresher" probability="1" requireddurationrange="1,2" requiredproximity="1" requirediscovered="false" />
      </ChangeTo>
```

![image](https://github.com/user-attachments/assets/272dd58d-e6d8-4c4d-aaf8-10d6cff0f3da)

In the second part of the video, the Survey is supposed to change into a "None" location (not an outpost so it doesnt turn into abandonedoutpost), but doesnt actually trigger until jumping *past it*

```xml
      <ChangeTo type="None" locationtypechangecooldown="2" messagetag="locationchange.outpost.changeto.abandoned" requirediscovered="false">
        <Requirement requiredlocations="Lair,LairMudraptor,LairThresher" probability="1" requireddurationrange="1,2" requiredproximity="1" requirediscovered="false" />
      </ChangeTo>
```

![image](https://github.com/user-attachments/assets/cb1d1b73-4fee-4eb1-bf75-e0f85ac8f8d8)

https://cdn.discordapp.com/attachments/1200592925382422530/1264158049527005235/Desktop_2023.09.10_-_17.38.09.06_52.mp4?ex=669cda6d&is=669b88ed&hm=eee0ca2069ef63af04ca5185e36cf350315b977bdfc3847f759bb0624de7b2eb&

Here's a good final example:
On the left, the location changed, on the right the location didnt change, despite both being visible

![image](https://github.com/user-attachments/assets/56b6e023-2fa7-408d-a682-24d773981d56)

### Reproduction steps

1. Use Dynamic Europa https://steamcommunity.com/sharedfiles/filedetails/?id=2532991202
2. Start Campaign
3. `togglecampaignteleport`
4. Try jumping near until a lair is visible (there's one in each biome)
5. Jump around  to try and trigger the Changeto on adjacent outposts (bug)
6. Jump past lairs and see locations actually turn

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
> After jumping using cheats (I did this multiple times previously), the location does not turn until I jump past it, pushing back the fog of war

This is intentional: locations don't change their type if they're further on the map than the furthest location you've travelled to.

**Reply by ubertpendragon** [reply]
That's a bit of a problem with what we're trying to do unfortunately 😔

**Reply by SomeRandomNoobKekeke** [reply]
In theory this could be "fixed" with 2 lines of lua
by assigning end location to furthestDiscoveredLocation before [ProgressWorld](https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Map/Map/Map.cs#L1210)
![barodev small](https://github.com/user-attachments/assets/616b8902-3442-4c8a-aec1-55872fbcf2f2)
